main.tsp

```diff
@@ -1,278 +0,0 @@
-
-  /** When the todo item was makred as completed */
-  @visibility("read") completedAt: utcDateTime;
-
-  // Want the read form to be normalized to TodoLabelRecord[], but can't
-  // https://github.com/microsoft/typespec/issues/2926
-  labels: TodoLabel[];
-}
-
-union TodoLabel {
-  string,
-  string[],
-  TodoLabelRecord,
-  TodoLabelRecord[],
-}
-
-model TodoLabelRecord {
-  name: string;
-
-  @format("^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$")
-  color?: string;
-}
-
-union TodoAttachment {
-  file: TodoFileAttachment,
-  url: TodoUrlAttachment,
-}
-
-model TodoUrlAttachment {
-  /** A description of the URL */
-  description: string;
-
-  /** The url */
-  url: url;
-}
-
-model TodoFileAttachment {
-  /** The todo item this is attached to */
-  todoItemId: TodoItem.id;
-
-  /** The file name of the attachment */
-  @maxLength(255)
-  filename: string;
-
-  /** The media type of the attachment */
-  mediaType: string;
-
-  /** The url where the attachment can be downloaded from */
-  url: url;
-}
-
-enum TodoItemStatus {
-  NotStarted,
-  InProgress,
-  Completed,
-}
-
-@error
-model Error {
-  /** A machine readable error code */
-  code: string;
-
-  /** A human readable message */
-  message: string;
-}
-
-namespace Users {
-  // would prefer to extend
-  // https://github.com/microsoft/typespec/issues/2922
-
-  model UserCreatedResponse {
-    ...User;
-    ...OkResponse;
-
-    /** The token to use to construct the validate email address url */
-    token: string;
-  }
-
-  /** The user already exists */
-  model UserExistsResponse extends Error {
-    ...ConflictResponse;
-  }
-
-  /** The user is invalid (e.g. forgot to enter email address) */
-  model InvalidUserResponse extends Error {
-    @statusCode statusCode: 422;
-  }
-
-  @route("/users")
-  @post
-  op create(
-    user: User,
-  ): UserCreatedResponse | UserExistsResponse | InvalidUserResponse;
-
-  @route("validate")
-  @get
-  op validate(@query token: string): OkResponse | InvalidUserResponse;
-
-  @route("login")
-  @post
-  op login(
-    username: User.username,
-    password: User.password,
-  ): OkResponse | UnauthorizedResponse;
-
-  @route("logout")
-  @get
-  op logout(): OkResponse;
-
-  /** Sends a reset token to the user's email address */
-  @route("forgot-password")
-  @post
-  op forgotPassword(email: User.email): OkResponse | NotFoundResponse;
-
-  @route("reset-password")
-  @get
-  op resetPassword(@query resetToken: string): OkResponse | NotFoundResponse;
-}
-
-@route("items")
-namespace TodoItems {
-  model PaginationControls {
-    /** The limit to the number of items */
-    @query limit: int32 = 50;
-
-    /** The offset to start paginating at */
-    @query offset: int32 = 0;
-  }
-
-  model TodoPage {
-    /** The items in the page */
-    items: TodoItem[];
-
-    /** The number of items returned in this page */
-    pageSize: int32;
-
-    /** The total number of items */
-    totalSize: int32;
-
-    ...PaginationControls;
```

Comment:
This feels odd from a modeling perspective. The PaginationControls only have things that are effectively write (request)-only, and the TodoPage model itself is read (response)-only. So these properties are erased from every point of usage, no?